lib/gc/ExecutionEngine/CPURuntime/Microkernel/BrgemmOnednn.cpp

```diff
@@ -137,24 +150,38 @@ void dnnl_brgemm_tilerelease() {
 void dnnl_brgemm_execute(int64_t kernel_idx, void *A, uint64_t A_offset,
                          void *B, uint64_t B_offset, void *C, uint64_t C_offset,
                          int num) {
+  auto it = tl_cache.find(kernel_idx);
   brgemm_kernel_t *kernel = nullptr;
-  size_t A_offset_in_bytes;
-  size_t B_offset_in_bytes;
-  size_t C_offset_in_bytes;
-  {
+  brgemm_desc_t *desc_ptr = nullptr;
+
+  if (it != tl_cache.end()) {
+    desc_ptr = &it->second.desc;
+    kernel = it->second.kernel;
+  } else {
     read_lock_guard_t g(g_brgemm_lock);
```

Comment:
Since it's thread local, do we still need this lock?

Reply:
when the target brgemm kernel is not found in thread_local cache, we still need to lock the global cache to get the target brgemm.